### PR TITLE
Enable backups if client side encryption is enabled

### PR DESCRIPTION
**Pull request by mamoit**
TL;DR: This PR enables the native android backup feature **only** if the backup is encrypted on the client side.

I was setting up [seedvault](https://github.com/seedvault-app/seedvault) on LineageOS when I realized that backups were not a thing on my device for OpenTracks only.
From my brief issue searching I found 2 issues, #346 and subsequently #710 I understand that this is a touchy subject.

Even if I don't completely align with the rational behind the decision to not implement this straight out of the box, I respect it.

So as @dennisguse [replied in the latter issue](https://github.com/OpenTracksApp/OpenTracks/issues/710#issuecomment-822656039), here goes a PR to try to reach a solution to which everyone can be content with.

Followed the [android documentation](https://developer.android.com/guide/topics/data/autobackup#define-device-conditions).

It has been over 10 years since I last did any Android development and I don't have an environment set up, this should be a pretty simple change (famous last words) so I'll test the build output from the CI :disappointed:
In the meantime I'll keep this marked as draft.

### CODE REVIEW

**Comment by mamoit** [comment]
Ok, so the CI only runs the `apk-build ` step on `main` :facepalm: 
Well.. time to setup and relearn how to use the android studio.

**Comment by mamoit** [comment]
Config backup and restore seem to work, but not the tracks themselves.
They're probably stored on a different path.

**Comment by dennisguse** [comment]
Tracks are stored in a Sqlite database. 
Path should be defined in DatabaseHelper IIRC.

**Comment by mamoit** [comment]
Thanks @dennisguse, that was all the help I needed :slightly_smiling_face:
I only had included `sharedpref` (SharedPreferences), and skipped the `database`.
Now a backup, uninstall (with data deletion), and subsequent install also restores the tracks.

Documentation is [here](https://developer.android.com/guide/topics/data/autobackup#XMLSyntax).
The full list of options is `file`, `database`, `sharedpref`, `external` and `root`.
I don't know if something else should be included besides `database` and `sharedpref`.

The `sharedpref` should not need to have the `requireFlags="clientSideEncryption"` since they don't really expose the tracks, but I'll keep it unless you want to relax this option.

There seems to be some indication in the doc where this should only be used [if targetting android 11 or lower](https://developer.android.com/guide/topics/data/autobackup#include-exclude-android-11), and an extra xml should be used [if targetting android 12 or higher](https://developer.android.com/guide/topics/data/autobackup#include-exclude-android-12).

I'm guessing that the configuration for android >=12 should also be included but I'd rather get a more insightful opinion before changing what seems  to work on my android 12.

**Comment by dennisguse** [comment]
@mamoit looks great (still had no time to re-read the docs).
One thing: can you also test this with a track that contains a marker with a picture?
The pictures are stored as files in the private storage, so those might not yet be included.

PS/ we can enable this functionality as there seems to be some need.
All the arguments are still valid, but I have relaxed my view ;)

**Comment by mamoit** [comment]
Indeed the pictures on markers are not kept in the backup.
I'll test the inclusion of the `domain="file"` after work (GMT).

**Comment by mamoit** [comment]
Tested the `file` domain, but after looking into the `file_paths.xml` I figured the pictures actually are stored as external files.

**Comment by dennisguse** [comment]
Cool!

Then the only thing missing is the Android 12+ configs, right?

@rgmf @pstorch I more or less changed my mind; having automated backups would be great. And if people use Google's cloud backup feature, it is their decision.
Does this make sense?

**Comment by pstorch** [comment]
@dennisguse if the backup is encrypted, I'm fine with it.

**Comment by dennisguse** [comment]
@mamoit ping :)

**Comment by mamoit** [comment]
I'm sorry, got caught up on life.
I will most likely not be able to update the PR today, will do it tomorrow the latest.

I won't have a way to test it as thouroughly though since the backup already seems to work as is 😐

**Comment by mamoit** [comment]
Ok, I was able to do the changes still today.
Tested it again on my Android 12 and backup/restore still works (including marker pictures).

Don't really know how to perform any further tests, this is the implementation to the best of my abilities and documentation interpretation skills.
Feel free to suggest better XML file naming :persevere:

**Comment by dennisguse** [comment]
@mamoit we could use Androids API level for naming :)

**Comment by dennisguse** [comment]
Looks good!
Will be in tomorrow's nightly.

I will create a new release in the upcoming days.

**Comment by mamoit** [comment]
@dennisguse is it normal for the fdroid release to take considerably longer than the Google play store to be shipped?
I can still only find the previous version there.

**Comment by dennisguse** [comment]
Yes and varies a lot... longest was 10 days - there is a crawler checking repos, then compilation and finally signing (needs a human) on an airgapped computer. 

That's why I use the nightly myself ;)
